### PR TITLE
Added subscriber ID and opt-ins to new order notification email.

### PR DESCRIPTION
**Pull request by juanlopez4691**
Related to task [STIM-SSO-SHOP: Checkout: Order notification email to customer service](https://app.asana.com/0/383242129844224/633179815868773)

Subtasks:
- [Output subscriber_id in new-order mail to shop admin](https://app.asana.com/0/0/732513948176627/f)
- [Output opt-ins (all) in new-order mail to shop admin](https://app.asana.com/0/0/732513948176628/f)

### CODE REVIEW

File: src/WooCommerce.php [hunk 1]
```diff
@@ -485,4 +485,26 @@ public static function woocommerce_save_account_details_required_fields(array $f
...
+   * @woocommerce_email_order_meta
+   */
+  public static function woocommerce_email_order_meta($order) {
+    $user_id = $order->get_user_id();
+    if ($subscriber_id = get_user_meta($user_id, 'subscriber_id', TRUE)) {
+      echo '<p><strong>' . __('Subscription ID:', PLUGIN::L10N) . '</strong> ' . $subscriber_id . '</p>';
+    }
+    $userinfo = get_user_meta($user_id, Plugin::USER_META_USERINFO, TRUE);
+    if (empty($userinfo['id'])) {
+      throw new \LogicException('Unable to build user info: Missing SSO ID.');
```
Comment (sun) [inline src/WooCommerce.php]: Did this work for you in manual testing?

I'm afraid, at the time a user/visitor makes the purchase, the user is registered in the SSO, but our client application does not have the mapping yet.

Checking for the 'id' should only succeed after a user manually logged in through the front-end login form with their own username and password, which is the only case (right now) in which the openid-connect plugin retrieves the new/current userinfo from the SSO.

At the time of a checkout/purchase, there is no userinfo right now.  So for the initial order email, you can only consult the POST data of the order, but not the userinfo.

**Review by sun — CHANGES_REQUESTED** [review]

**Comment by sun** [comment]
@juanlopez4691 This branch has been rebased onto user-info-sun to resolve conflicts.

**Comment by juanlopez4691** [comment]
@sun Pushed a fixup. Thanks for your review.

File: src/WooCommerce.php [hunk 2]
```diff
@@ -575,14 +575,16 @@ public static function woocommerce_email_order_meta($order) {
     if ($subscriber_id = get_user_meta($user_id, 'subscriber_id', TRUE)) {
       echo '<p><strong>' . __('Subscription ID:', PLUGIN::L10N) . '</strong> ' . $subscriber_id . '</p>';
     }
-    $userinfo = get_user_meta($user_id, Plugin::USER_META_USERINFO, TRUE);
-    if (empty($userinfo['id'])) {
-      throw new \LogicException('Unable to build user info: Missing SSO ID.');
+    if (!isset($_POST['optins'])) {
```
Comment (sun) [inline src/WooCommerce.php]: `$_POST['optins']` does not exist; how did this work for you…?

Comment (juanlopez4691) [reply src/WooCommerce.php]: > At the time of a checkout/purchase, there is no userinfo right now. So for the initial order email, you can only consult the POST data of the order, but not the userinfo.

Obviously I am misunderstanding this.

**Comment by sun** [comment]
Rebased against latest user-info-sun.

**Comment by sun** [comment]
Merged into user-info-sun and will be enhanced in there.

**Pull request closed without merging**